Result of `nixpkgs-review pr 253711` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>32 packages built:</summary>
  <ul>
    <li>indexed_bzip2 (python310Packages.indexed_bzip2)</li>
    <li>indexed_bzip2.dist (python310Packages.indexed_bzip2.dist)</li>
    <li>python310Packages.indexed_gzip</li>
    <li>python310Packages.indexed_gzip.dist</li>
    <li>python310Packages.indexed_zstd</li>
    <li>python310Packages.indexed_zstd.dist</li>
    <li>python310Packages.lzmaffi</li>
    <li>python310Packages.lzmaffi.dist</li>
    <li>python310Packages.python-xz</li>
    <li>python310Packages.python-xz.dist</li>
    <li>rapidgzip (python310Packages.rapidgzip)</li>
    <li>rapidgzip.dist (python310Packages.rapidgzip.dist)</li>
    <li>ratarmount (python310Packages.ratarmount)</li>
    <li>ratarmount.dist (python310Packages.ratarmount.dist)</li>
    <li>python310Packages.ratarmountcore</li>
    <li>python310Packages.ratarmountcore.dist</li>
    <li>python311Packages.indexed_bzip2</li>
    <li>python311Packages.indexed_bzip2.dist</li>
    <li>python311Packages.indexed_gzip</li>
    <li>python311Packages.indexed_gzip.dist</li>
    <li>python311Packages.indexed_zstd</li>
    <li>python311Packages.indexed_zstd.dist</li>
    <li>python311Packages.lzmaffi</li>
    <li>python311Packages.lzmaffi.dist</li>
    <li>python311Packages.python-xz</li>
    <li>python311Packages.python-xz.dist</li>
    <li>python311Packages.rapidgzip</li>
    <li>python311Packages.rapidgzip.dist</li>
    <li>python311Packages.ratarmount</li>
    <li>python311Packages.ratarmount.dist</li>
    <li>python311Packages.ratarmountcore</li>
    <li>python311Packages.ratarmountcore.dist</li>
  </ul>
</details>


---

<details>
</summary>result of `$ fd . results/ --type x -L | sort`</summary>

```
results/indexed_bzip2/bin/ibzip2
results/indexed_bzip2/lib/python3.10/site-packages/indexed_bzip2.cpython-310-x86_64-linux-gnu.so
results/python310Packages.indexed_gzip/lib/python3.10/site-packages/indexed_gzip/indexed_gzip.cpython-310-x86_64-linux-gnu.so
results/python310Packages.indexed_gzip/lib/python3.10/site-packages/indexed_gzip/tests/ctest_indexed_gzip.cpython-310-x86_64-linux-gnu.so
results/python310Packages.indexed_gzip/lib/python3.10/site-packages/indexed_gzip/tests/ctest_zran.cpython-310-x86_64-linux-gnu.so
results/python310Packages.indexed_zstd/lib/python3.10/site-packages/indexed_zstd.cpython-310-x86_64-linux-gnu.so
results/python310Packages.lzmaffi/lib/python3.10/site-packages/_lzmaffi_mods/_compiled_module.cpython-310-x86_64-linux-gnu.so
results/python311Packages.indexed_bzip2/bin/ibzip2
results/python311Packages.indexed_bzip2/lib/python3.11/site-packages/indexed_bzip2.cpython-311-x86_64-linux-gnu.so
results/python311Packages.indexed_gzip/lib/python3.11/site-packages/indexed_gzip/indexed_gzip.cpython-311-x86_64-linux-gnu.so
results/python311Packages.indexed_gzip/lib/python3.11/site-packages/indexed_gzip/tests/ctest_indexed_gzip.cpython-311-x86_64-linux-gnu.so
results/python311Packages.indexed_gzip/lib/python3.11/site-packages/indexed_gzip/tests/ctest_zran.cpython-311-x86_64-linux-gnu.so
results/python311Packages.indexed_zstd/lib/python3.11/site-packages/indexed_zstd.cpython-311-x86_64-linux-gnu.so
results/python311Packages.lzmaffi/lib/python3.11/site-packages/_lzmaffi_mods/_compiled_module.cpython-311-x86_64-linux-gnu.so
results/python311Packages.rapidgzip/bin/rapidgzip
results/python311Packages.rapidgzip/lib/python3.11/site-packages/rapidgzip.cpython-311-x86_64-linux-gnu.so
results/python311Packages.ratarmount/bin/ratarmount
results/rapidgzip/bin/rapidgzip
results/rapidgzip/lib/python3.10/site-packages/rapidgzip.cpython-310-x86_64-linux-gnu.so
results/ratarmount/bin/ratarmount
```
</details>


---

It seems the tests are not being run, but they spend time trying to do so.
Consider using `pytestCheckHook` if the tests are present.
If no tests are distributed on Pypi then it is quite common to instead pull the package from github.
If no tests exist at all, then please either configure `doCheck = false;` with a comment stating there are no tests, or of something major is prohibiting running the tests.

<details>
<summary>result of `$ rg 'Ran 0 tests' logs`</summary>

```
$ rg 'Ran 0 tests' logs
logs/indexed_bzip2.log
184:Ran 0 tests in 0.000s

logs/indexed_bzip2.dist.log
184:Ran 0 tests in 0.000s

logs/python310Packages.indexed_gzip.log
281:Ran 0 tests in 0.000s

logs/python310Packages.indexed_gzip.dist.log
281:Ran 0 tests in 0.000s

logs/python310Packages.indexed_zstd.log
183:Ran 0 tests in 0.000s

logs/python310Packages.indexed_zstd.dist.log
183:Ran 0 tests in 0.000s

logs/python310Packages.lzmaffi.log
224:Ran 0 tests in 0.000s

logs/python310Packages.lzmaffi.dist.log
224:Ran 0 tests in 0.000s

logs/python310Packages.python-xz.dist.log
208:Ran 0 tests in 0.000s

logs/python310Packages.python-xz.log
208:Ran 0 tests in 0.000s

logs/rapidgzip.log
324:Ran 0 tests in 0.000s

logs/rapidgzip.dist.log
324:Ran 0 tests in 0.000s

logs/python311Packages.indexed_bzip2.log
184:Ran 0 tests in 0.000s

logs/python311Packages.indexed_bzip2.dist.log
184:Ran 0 tests in 0.000s

logs/python311Packages.indexed_gzip.log
281:Ran 0 tests in 0.000s

logs/python311Packages.indexed_gzip.dist.log
281:Ran 0 tests in 0.000s

logs/python311Packages.indexed_zstd.log
183:Ran 0 tests in 0.000s

logs/python311Packages.indexed_zstd.dist.log
183:Ran 0 tests in 0.000s

logs/python311Packages.lzmaffi.log
224:Ran 0 tests in 0.000s

logs/python311Packages.python-xz.log
208:Ran 0 tests in 0.000s

logs/python311Packages.python-xz.dist.log
208:Ran 0 tests in 0.000s

logs/python311Packages.rapidgzip.log
324:Ran 0 tests in 0.000s

logs/python311Packages.lzmaffi.dist.log
224:Ran 0 tests in 0.000s

logs/python311Packages.rapidgzip.dist.log
324:Ran 0 tests in 0.000s
```
</details>

---


- [-] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [ ] executables tested on x86_64-linux
  * `ratarmount` has an error, shown below
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
  * I even verified xz was properly referenced in the outputs of lzmaffi
- [x] source is fetched using the appropriate function
  * Arguably `fetchFromGitHub` is better if there are tests.
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored



---

I get the following error when i try to run `ratarmount`:

```
$ results/ratarmount/bin/ratarmount
Traceback (most recent call last):
  File "/nix/store/fb2m5jg4sjf7wg7qpb7zj0rkr150lvwx-python3.10-ratarmount-0.14.0/bin/.ratarmount-wrapped", line 6, in <module>
    from ratarmount import cli
  File "/nix/store/2hzyh44p871vvk8iy1h3nkbykdxsc9b2-python3.11-ratarmount-0.14.0/lib/python3.11/site-packages/ratarmount.py", line 35, in <module>
    import ratarmountcore as core
  File "/nix/store/ygkwzv0kmd79dkd0kzjd9420kyw2h0ri-python3.11-ratarmountcore-0.6.0/lib/python3.11/site-packages/ratarmountcore/__init__.py", line 47, in <module>
    from .compressions import (
  File "/nix/store/ygkwzv0kmd79dkd0kzjd9420kyw2h0ri-python3.11-ratarmountcore-0.6.0/lib/python3.11/site-packages/ratarmountcore/compressions.py", line 29, in <module>
    import lzmaffi
  File "/nix/store/w1afx5wpg7harndqpbpf0249jnpmx42g-python3.11-lzmaffi-0.3.0/lib/python3.11/site-packages/lzmaffi/__init__.py", line 29, in <module>
    from ._lzmamodule2 import *
  File "/nix/store/w1afx5wpg7harndqpbpf0249jnpmx42g-python3.11-lzmaffi-0.3.0/lib/python3.11/site-packages/lzmaffi/_lzmamodule2.py", line 278, in <module>
    m = ffi.verify("""
  File "/nix/store/s5gspkfm4skl2lcx40q6vz05fv3yxn3x-python3.11-cffi-1.15.1/lib/python3.11/site-packages/cffi/api.py", line 468, in verify
    lib = self.verifier.load_library()
  File "/nix/store/s5gspkfm4skl2lcx40q6vz05fv3yxn3x-python3.11-cffi-1.15.1/lib/python3.11/site-packages/cffi/verifier.py", line 104, in load_library
    self._write_source()
  File "/nix/store/s5gspkfm4skl2lcx40q6vz05fv3yxn3x-python3.11-cffi-1.15.1/lib/python3.11/site-packages/cffi/verifier.py", line 193, in _write_source
    with open(self.sourcefilename, "w") as fp:
OSError: [Errno 30] Read-only file system: '/nix/store/w1afx5wpg7harndqpbpf0249jnpmx42g-python3.11-lzmaffi-0.3.0/lib/python3.11/site-packages/lzmaffi/__pycache__/_compiled_module.c'
```

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -11945,6 +11945,11 @@
     githubId = 59313755;
     name = "Maxim Karasev";
   };
+  mxmlnkn = {
+    github = "mxmlnkn";
+    githubId = 6842824;
+    name = "Maximilian Knespel";
+  };
   myaats = {
     email = "mats@mats.sh";
     github = "Myaats";

--- a/pkgs/development/python-modules/indexed-bzip2/default.nix
+++ b/pkgs/development/python-modules/indexed-bzip2/default.nix
@@ -1,0 +1,31 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "indexed_bzip2";
+  version = "1.5.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-tKf9odadfQZQYJz//vWYpeB99Z8VLg+hEPvfEHXgdnM=";
+  };
+
+  # has no tests
+  doCheck = false;
+
+  pythonImportsCheck = [ "indexed_bzip2" ];
+
+  meta = with lib; {
+    description = "Python library for parallel decompression and seeking within compressed bzip2 files";
+    homepage = "https://github.com/mxmlnkn/indexed_bzip2";
+    license = licenses.mit; # dual MIT and asl20, https://internals.rust-lang.org/t/rationale-of-apache-dual-licensing/8952
+    maintainers = with lib.maintainers; [ mxmlnkn ];
+    platforms = platforms.all;
+  };
+}

--- a/pkgs/development/python-modules/indexed-gzip/default.nix
+++ b/pkgs/development/python-modules/indexed-gzip/default.nix
@@ -1,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, cython
+, zlib
+}:
+
+buildPythonPackage rec {
+  pname = "indexed_gzip";
+  version = "1.8.5";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-h9JgYq9KxmknaRuMgq+5YWA8tUaFk+lclkqdNAnr/cI=";
+  };
+
+  nativeBuildInputs = [ cython ];
+
+  buildInputs = [ zlib ];
+
+  # Too complicated to get to work, not a simple pytest call.
+  doCheck = false;
+
+  pythonImportsCheck = [ "indexed_gzip" ];
+
+  meta = with lib; {
+    description = "Python library to seek within compressed gzip files";
+    homepage = "https://github.com/pauldmccarthy/indexed_gzip";
+    license = licenses.zlib;
+    maintainers = with lib.maintainers; [ mxmlnkn ];
+    platforms = platforms.all;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -9354,6 +9354,8 @@ with pkgs;
 
   incron = callPackage ../tools/system/incron { };
 
+  indexed-bzip2 = with python3Packages; toPythonApplication indexed-bzip2;
+
   industrializer = callPackage ../applications/audio/industrializer { };
 
   inetutils = callPackage ../tools/networking/inetutils { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -5178,6 +5178,8 @@ self: super: with self; {
 
   indexed-bzip2 = callPackage ../development/python-modules/indexed-bzip2 { };
 
+  indexed-gzip = callPackage ../development/python-modules/indexed-gzip { inherit (pkgs) zlib; };
+
   infinity = callPackage ../development/python-modules/infinity { };
 
   inflect = callPackage ../development/python-modules/inflect { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -5176,6 +5176,8 @@ self: super: with self; {
 
   incremental = callPackage ../development/python-modules/incremental { };
 
+  indexed-bzip2 = callPackage ../development/python-modules/indexed-bzip2 { };
+
   infinity = callPackage ../development/python-modules/infinity { };
 
   inflect = callPackage ../development/python-modules/inflect { };

